portal-impl/src/com/liferay/portal/service/impl/RegionLocalServiceImpl.java

```
@@ -68,78 +68,109 @@ public void deleteCountryRegions(long countryId) {
 	}
 
 	@Override
-	public Region deleteRegion(long regionId) {
+	public Region deleteRegion(long regionId) throws PortalException {
 		Region region = regionPersistence.fetchByPrimaryKey(regionId);
```

We should use `findByPrimaryKey` to throw an exception if regionId is invalid.